### PR TITLE
clear and save/print buttons

### PR DESCRIPTION
**Pull request by aliezz19**
Implemented the clear and save/print buttons to function.

### CODE REVIEW

**Review by shiraliyeva — APPROVED** [review]
Everything looks good, but there are some CA violations! 
Have a look at where you are calling the DAO.
Also the Controllers in TextAreaView should be called in a different way, but we can fix that easily after we merge with the other features.

File: src/interface_adapter/save_text/SavePresenter.java [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
         this.saveViewModel = saveViewModel;
     }
 
-    public void success(SaveInputData inputData) {
-        this.saveViewModel.firePropertyChanged();
+    @Override
+    public void prepareSaveView(SaveOutputData outputData) {
+        this.saveViewModel.setContent(outputData.getOutputText());
+        final SaveDataAccessObject saveDataAccessObject = new SaveDataAccessObject(outputData.getOutputText());
+        SaveDataAccessObject.saveToPDF(outputData.getOutputText());
```
Comment (nothing128) [inline src/interface_adapter/save_text/SavePresenter.java:22]: Data Access Objects should only be called in Main

File: src/view/TextAreaView.java [hunk 2]
```diff
@@ -46,12 +56,46 @@ public TextAreaView(TextAreaViewModel textAreaViewModel) {
 
         // Add the mainPanel to the CENTER of TextAreaView
         this.add(mainPanel, BorderLayout.CENTER);
+
+        clearButton.addActionListener(new ActionListener() {
+            final ClearViewModel clearViewModel = new ClearViewModel();
+
+            final ClearOutputBoundary clearOutputBoundary = new ClearPresenter(clearViewModel, textArea);
```
Comment (nothing128) [inline src/view/TextAreaView.java:63]: ClearPresenter and ClearInteractor cannot be directly created from TextAreaView as it violates CA

File: src/view/TextAreaView.java [hunk 3]
```diff
@@ -46,12 +56,46 @@ public TextAreaView(TextAreaViewModel textAreaViewModel) {
...
+            }
+        });
+
+        saveButton.addActionListener(new ActionListener() {
+
+            final SaveViewModel saveViewModel = new SaveViewModel();
+
+            final SaveOutputBoundary saveOutputBoundary = new SavePresenter(saveViewModel);
+
+            final SaveInputBoundary saveUseCaseInteractor = new SaveInteractor(saveOutputBoundary);
```
Comment (nothing128) [inline src/view/TextAreaView.java:82]: SavePresenter and SaveInteractor cannot be directly created in View as it violates CA.

**Review by nothing128 — CHANGES_REQUESTED** [review]
Nice work! Most of the code is fine, but a bit of it violates CA.

- [x] In SavePresenter, move the DataAccessObject call to main
- [x] In textAreaView, ClearPresenter and ClearInteractor should not directly Interact with View
- [x] The same issue exists with SavePresenter and ClearInteractor

**Review by shiraliyeva — APPROVED** [review]
Great progress! Looks good, just one minor change i would suggest. Since we are not using ClearViewModel anymore, I think you may omit that from ClearPresenter.

**Review by nothing128 — APPROVED** [review]
Looks good!

**Pull request merged**